### ISSUE TITLE
Crash with corrupted scenery due to de-selected scenery theme

### ISSUE THREAD

**Issue by MaxMallon** [OpenRCT2/OpenRCT2#16436]
### Operating System

Windows 10, 64bit

### OpenRCT2 build

OpenRCT2, v0.3.5.1 (6e839bd on develop) provided by GitHub

### Describe the issue

I have a park with a lot of scenery selected (custom scenery too), and my scenery theme tabs were getting out of hand. So, I opted to remove a few themes I didn't need anymore, expecting the items I had previously placed to remain selected and appear under the ? theme. Apparently, that did not happen for everything. 
Specifically, the Wild West houses I had placed, remained, and turned invisible. I didn't notice at first, and wanted to build something there, which wasn't possible. With disabled clearance checks it's possible to build small scenery on the seemingly empty piece of grass, but when I went to place a large scenery item, the game instantly crashed, no message or anything, just boom gone. 
Not knowing yet what was going on, I went to the tile inspector, and noticed large scenery being there. Trying to remove it crashed the game in the same manner. Only later did I realize I'd placed those houses there, enabling the theme again made the houses reappear, and allowed 2 of the 3 to be removed without issue, the 3rd one still crashed the game though.. It is Large Scenery Type 197, which oddly enough does not corrospond to any of the 3 Wild West buildings. I don't know what it could be then. 

Point is, it shouldn't crash when trying to either place or delete scenery that is unknown, and it shouldn't untick items that are already used in the park. 
I can't reproduce it.

### Area(s) with issue?

New Save Format (NSF)

### Steps to reproduce

Can't reproduce it, but this is the general steps that happened:
1: have a park with a ton of scenery selected
2: place large item from scenery in park (more than 1 tile)
3: unload the theme the object belonged to (with object selection)
4: notice the scenery disappearing
5: use tile inspector to remove scenery, or attempt to place other large scenery in the same place
6: observe crash

**Comment by Gymnasiast** [comment]
Preventing crashes when the objects are not loaded is tricky (although we try). You’re right about not unticking items that are actually used, though.

**Comment by MaxMallon** [comment]
Now that I have the codebase pulled, I simply tried the save again, and found the fix for at least the crash. 
Should I commit on my fork, and do a pull request? Or does it need to undergo tests etc?
It's literally adding a null-check in 1 line, specifically, line 460 of MapAnimation.cpp

**Comment by duncanspumpkin** [comment]
make a pull request of your change and we can review it.

**Comment by James103** [comment]
> make a pull request of your change and we can review it.

See: #16462

**Comment by Gymnasiast** [comment]
This is not fixed, it shouldn’t unload used objects in the first place. What has been fixed is one of the symptoms if that happens anyway.

**Comment by Broxzier** [comment]
The fix just prevents crashing when you open a corrupt park. To properly fix this, the objects should be readded or removed on import, and preventing this from happening again. Then the null-check fix could even be removed again.

**Comment by duncanspumpkin** [comment]
@MaxMallon we aren't saying we would remove the check. Gymnasiast was just providing context as to why he was reopening it after I closed it.

**Comment by duncanspumpkin** [comment]
The problem seems to be due to placing scenery whilst you have the object selection window open. The object selection window only calculates if an object has been placed and is in use when you first open the window. To prevent this from causing issues either the window needs to be recalculated when placing scenery or alternatively there should be an additional check when unloading an object or both.

**Comment by MaxMallon** [comment]
I have a fix for that part of the issue too now, thing is, I'm not sure if there might be side effects. I don't think so, but who knows. Should I just do another pull request?

**Comment by Gymnasiast** [comment]
Yeah, open the pull request, but make sure to mention it needs testing. It doesn’t hurt either to do as many tests yourself of side effects and list what you tested.
